src/cart/src/cart/crt_ctl.c

```diff
@@ -154,12 +125,7 @@ crt_hdlr_ctl_get_uri_cache(crt_rpc_t *rpc_req)
 	if (rc != 0)
 		D_GOTO(out, rc);
 
-	rc = d_hash_table_traverse(&grp_priv->gp_uri_lookup_cache,
-				   crt_ctl_get_uri_cache_size_cb, &nuri);
-	if (rc != 0)
-		D_GOTO(out, 0);
-
-	D_ALLOC_ARRAY(uri_cache.grp_cache, nuri);
+	D_ALLOC_ARRAY(uri_cache.grp_cache, CRT_SRV_CONTEXT_NUM);
```

Comment:
I'm having trouble following this code, it's traverersing a hash table, and copying in up to CRT_SRV_CONTEXT_NUM per hash-table entry by the looks of it, so this buffer could potentialy be over-run?